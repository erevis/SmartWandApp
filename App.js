import * as React from 'react';
import {StyleSheet, View, Text, TextInput, Button} from 'react-native'
import { createMaterialBottomTabNavigator } from '@react-navigation/material-bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import 'react-native-gesture-handler';
import { Entypo , AntDesign , MaterialIcons } from '@expo/vector-icons';
import {Therapy} from './components/Therapy.js'
import {Education} from './components/Education.js'
import {Progress} from './components/Progress.js'
import {Account} from './components/Account.js'
import AuthContext from './components/AuthContext.js';
import * as SecureStore from 'expo-secure-store';

const Tab = createMaterialBottomTabNavigator();

function Tabs() {
  return (
    <Tab.Navigator
      initialRouteName="Therapy"
      activeColor="#e9cb1e"
      labelStyle={{ fontSize: 12 }}
      barStyle={{ backgroundColor: '#202c40' , paddingBottom: 0}}
    >
      <Tab.Screen
        name="Therapy"
        component={Therapy}
        options={{
          tabBarLabel: 'Therapy',
          tabBarIcon: ({ color }) => (
            <AntDesign name="medicinebox" size={24} color={color} />
          ),
        }}
      />
      <Tab.Screen
        name="Education"
        component={Education}
        options={{
          tabBarLabel: 'Education',
          tabBarIcon: ({ color }) => (
            <AntDesign name="book" size={24} color={color} />
          ),
        }}
      />
      <Tab.Screen
        name="Progress"
        component={Progress}
        options={{
          tabBarLabel: 'Progress',
          tabBarIcon: ({ color }) => (
            <Entypo name="progress-two" size={24} color={color} />
          ),
        }}
      />
      <Tab.Screen
        name="Account"
        component={Account}
        options={{
          tabBarLabel: 'Account',
          tabBarIcon: ({ color }) => (
            <MaterialIcons name="account-circle" size={24} color={color}/> 
          ),
        }}
      />
    </Tab.Navigator>
  );
}

const Stack = createStackNavigator();

function SignInScreen() {
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');

  const { signIn } = React.useContext(AuthContext);

  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Username"
        value={username}
        style={styles.input}
        onChangeText={setUsername}
      />
      <TextInput
        placeholder="Password"
        value={password}
        style={styles.input}
        onChangeText={setPassword}
        secureTextEntry
      />
      <Button 
      title="Sign in" 
      style={styles.input}
      onPress={() => signIn({ username, password })} />
    </View>
  );
}

export default function App() {
  const [state, dispatch] = React.useReducer(
    (prevState, action) => {
      switch (action.type) {
        case 'RESTORE_TOKEN':
          return {
            ...prevState,
            userToken: action.token,
            isLoading: false,
          };
        case 'SIGN_IN':
          return {
            ...prevState,
            isSignout: false,
            userToken: action.token,
          };
        case 'SIGN_OUT':
          return {
            ...prevState,
            isSignout: true,
            userToken: null,
          };
      }
    },
    {
      isLoading: true,
      isSignout: false,
      userToken: null,
    }
  )
  // return (
  //     <NavigationContainer>
  //         <MyStack />
  //     </NavigationContainer>
  // );

  React.useEffect(() => {
    // Fetch the token from storage then navigate to our appropriate place
    const bootstrapAsync = async () => {
      let userToken;

      try {
        userToken = await SecureStore.getItemAsync('userToken');
      } catch (e) {
        // Restoring token failed
      }

      // After restoring token, we may need to validate it in production apps

      // This will switch to the App screen or Auth screen and this loading
      // screen will be unmounted and thrown away.
      dispatch({ type: 'RESTORE_TOKEN', token: userToken });
    };

    bootstrapAsync();
  }, []);

  const authContext = React.useMemo(
    () => ({
      signIn: async data => {
        // In a production app, we need to send some data (usually username, password) to server and get a token
        // We will also need to handle errors if sign in failed
        // After getting token, we need to persist the token using `SecureStore`
        // In the example, we'll use a dummy token

        dispatch({ type: 'SIGN_IN', token: 'dummy-auth-token' });
      },
      signOut: () => dispatch({ type: 'SIGN_OUT' }),
      signUp: async data => {
        // In a production app, we need to send user data to server and get a token
        // We will also need to handle errors if sign up failed
        // After getting token, we need to persist the token using `SecureStore`
        // In the example, we'll use a dummy token

        dispatch({ type: 'SIGN_IN', token: 'dummy-auth-token' });
      },
    }),
    []
  );

  return (
    <AuthContext.Provider value={authContext}>
      <NavigationContainer>
        <Stack.Navigator>
          {state.userToken == null ? (
            <Stack.Screen
              name="SignIn"
              component={SignInScreen}
              options={{
                title: 'Smart Wand Companion App',
                headerTitleAlign: 'center',
                animationTypeForReplace: state.isSignout ? 'pop' : 'push',
                headerStyle: {
                  backgroundColor: '#4fe0dc',
                },
                headerTintColor: '#fff',
                headerTitleStyle: {
                  fontWeight: 'bold',
                }
              }}
            />
          ) : (
            <Stack.Screen
            name="Home"
            component={Tabs}
            options={{headerShown: false}}
            />
          )}
        </Stack.Navigator>
      </NavigationContainer>
    </AuthContext.Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ecf0f1',
    alignItems: 'center',
    justifyContent: 'center',
  },
  input: {
    width: 200,
    height: 40,
    padding: 10,
    borderWidth: 1,
    borderColor: 'black',
    marginBottom: 10,
  }
});