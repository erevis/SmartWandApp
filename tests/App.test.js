import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

import App, {Tabs} from '../App.js';


describe('<App />', () => {
    act(() => {
        it('Renders correctly', () => {
            //expect.assertions(1)
            const tree = renderer.create(<App />).toJSON();
            //console.log(tree)
            expect(tree).toMatchSnapshot();
        });
    })
    
    act(() => {
        it('Has 1 child', () => {
            const tree = renderer.create(<App />).toJSON();
            console.log(tree)
            expect(tree.children.length).toBe(1);
        });
    })

    // act(() => {
    //     it('Navigation', ()=> {
            
    //         const nav = renderer.create(
    //             <NavigationContainer>
    //                 <Tabs />
    //             </NavigationContainer>
    //         );
    //         console.log(nav)
    //     })
    // })
})