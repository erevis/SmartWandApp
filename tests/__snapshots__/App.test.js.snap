// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#d8b53d",
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#476971",
                              "borderRadius": 86,
                              "height": 1334,
                              "left": -400.2,
                              "position": "absolute",
                              "top": -1000.5,
                              "transform": Array [
                                Object {
                                  "rotate": "35deg",
                                },
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                              ],
                              "width": 1334,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                              ],
                            }
                          }
                        >
                          <RNSVGSvgView
                            bbHeight={350}
                            bbWidth={750}
                            focusable={false}
                            height={350}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                Object {
                                  "flex": 0,
                                  "height": 350,
                                  "width": 750,
                                },
                              ]
                            }
                            width={750}
                          >
                            <RNSVGGroup>
                              <RNSVGPath
                                d="M-20 378.5C31.5 32.5 302.5 463 410 0C447.5 -285 375 644 500 644H0C0 644 -66.5 724.5 -17.5 378.5Z"
                                fill={4282870129}
                                propList={
                                  Array [
                                    "fill",
                                    "stroke",
                                  ]
                                }
                                stroke={4282870129}
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                        <View
                          style={
                            Object {
                              "backgroundColor": "#476971",
                              "borderRadius": 86,
                              "bottom": -133.4,
                              "height": 1334,
                              "left": -1467.4,
                              "position": "absolute",
                              "transform": Array [
                                Object {
                                  "rotate": "40deg",
                                },
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                              ],
                              "width": 1334,
                            }
                          }
                        />
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 250,
                              "marginBottom": 40,
                              "marginTop": 60,
                              "width": 250,
                            }
                          }
                        />
                        <TextInput
                          allowFontScaling={true}
                          fontWeight="bold"
                          onChangeText={[Function]}
                          placeholder="Email"
                          placeholderTextColor="#476971"
                          rejectResponderTermination={true}
                          style={
                            Object {
                              "backgroundColor": "#ffe4e0",
                              "borderColor": "black",
                              "borderRadius": 25,
                              "borderWidth": 0,
                              "color": "#FFFFFF",
                              "height": 50,
                              "marginBottom": 15,
                              "padding": 10,
                              "paddingLeft": 20,
                              "width": "70%",
                            }
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                        <TextInput
                          allowFontScaling={true}
                          fontWeight="bold"
                          onChangeText={[Function]}
                          placeholder="Password"
                          placeholderTextColor="#476971"
                          rejectResponderTermination={true}
                          secureTextEntry={true}
                          style={
                            Object {
                              "backgroundColor": "#ffe4e0",
                              "borderColor": "black",
                              "borderRadius": 25,
                              "borderWidth": 0,
                              "color": "#FFFFFF",
                              "height": 50,
                              "marginBottom": 15,
                              "padding": 10,
                              "paddingLeft": 20,
                              "width": "70%",
                            }
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "height": 30,
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontStyle": "italic",
                              }
                            }
                          >
                            Forgot Password?
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ec6767",
                              "borderRadius": 25,
                              "height": 50,
                              "justifyContent": "center",
                              "marginTop": 40,
                              "opacity": 1,
                              "width": "70%",
                            }
                          }
                        >
                          <Text
                            style={Object {}}
                          >
                            LOGIN
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "marginTop": 15,
                              }
                            }
                          >
                            Signup
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
